src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_text.cs

```diff
@@ -68,91 +171,152 @@ public static unsafe bool TryParse(ref this BufferReader<byte> reader, out bool
             if (Utf8Parser.TryParse(unread, out value, out int consumed, standardFormat))
             {
                 reader.Advance(consumed);
-                return true;
+                return consumed;
+            }
+            else if (unread.Length >= MaxBoolLength)
```

Comment:
I am not sure if this is correct. What if `unread = {tru1}`? TryParse returns false, `unread.Length is < MaxBoolLength`, and then we fall to TryParseSlow which isn't what we want to do here.

Reply:
It is what we want to do, if we want this code path to be as fast as possible. Every check shows up in the perf runs. All of these methods make some technically unnecessary trips down the slow path.